tools/codegen/gentable.py

```diff
@@ -215,7 +223,16 @@ def generate(self, path, template="default"):
             for option in column.options:
                 # Only allow explicitly-defined options.
                 if option in COLUMN_OPTIONS:
-                    column_options.append("ColumnOptions::" + COLUMN_OPTIONS[option])
+                    if option == "collate":
+                        collation = str(column.options["collate"]).upper()
+                        if collation in COLLATIONS:
+                            column_options.append("ColumnOptions::" + COLUMN_OPTIONS[option] + collation)
+                        else:
+                            print(yellow(
+                                "Table %s column %s contains an unknown collation sequence: %s" % (
+                                self.table_name, column.name, column.options["collate"])))
```

Comment:
I suspect we should error out here. Not just print a warning

Reply:
I was matching the error output to the other column options, but I'm good with a full exit when incorrectly defined.